### ISSUE TITLE
Failed to connect Google Calendar: ValueError("time data does not match format")

### ISSUE THREAD

**Issue by mfreeman72** [slgobinath/gcalendar#2]
When I run gcalendar, the browser window comes up and I log in, click Allow, etc. After the process is over, the terminal window says the following:

```
/usr/lib/python3/dist-packages/oauth2client/_helpers.py:255: UserWarning: Cannot access /home/mike/.config/gcalendar/mikefreeman1972_v1.dat: No such file or directory
  warnings.warn(_MISSING_FILE_MESSAGE.format(filename))

Your browser has been opened to visit:

    https://accounts.google.com/o/oauth2/auth?client_id=1097830982036-5opi3qc4lje5ak0so1mjg0icktiiu289.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Flocalhost%3A48935%2F&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcalendar.readonly&access_type=offline&response_type=code

If your browser is on a different machine then exit and re-run this
application with the command-line parameter

  --noauth_local_webserver

Authentication successful.
Failed to connect Google Calendar
```
After this, if I activate the Google Calendar desklet, nothing appears on the screen. Looking in the configuration window for the desklet, everything that looks like it should have some value or information in it is blank. This is on Linux Mint 20.

**Comment by slgobinath** [comment]
`Failed to connect Google Calendar` looks like something went wrong in the authentication process. Without solving it the Cinnamon desklet won't work.

Do you have proxy/firewall/XDM running on your machine?

**Comment by slgobinath** [comment]
Could you please try the following commands and let me know the error message:

```
git clone https://github.com/slgobinath/gcalendar.git
cd gcalendar
python3 -m gcalendar --debug
```

Thanks

**Comment by slgobinath** [comment]
As mentioned in this link https://discuss.python.org/t/parse-z-timezone-suffix-in-datetime/2220 use `dateutil.parser.isoparse` to parse rfc3339 date time format.

Note that Google Calendar return start and end date time in rfc3339 format.

**Comment by infomanPMF** [comment]
I've experienced the same error as mentioned above. 
Running
```git clone https://github.com/slgobinath/gcalendar.git
cd gcalendar
python3 -m gcalendar --debug
```
returned
```Failed to connect Google Calendar
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "~/gcalendar/gcalendar/__main__.py", line 201, in <module>
    main()
  File "~/gcalendar/gcalendar/__main__.py", line 195, in main
    raise ex
  File "~/gcalendar/gcalendar/__main__.py", line 179, in main
    g_calendar = GCalendar(client_id, client_secret, account_id, storage_path)
  File "~/gcalendar/gcalendar/gcalendar.py", line 38, in __init__
    self.service = self.create_service()
  File "~/gcalendar/gcalendar/gcalendar.py", line 65, in create_service
    return discovery.build("calendar", "v3", http=http)
  File "~/.local/lib/python3.6/site-packages/googleapiclient/_helpers.py", line 134, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "~/.local/lib/python3.6/site-packages/googleapiclient/discovery.py", line 264, in build
    raise e
  File "~/.local/lib/python3.6/site-packages/googleapiclient/discovery.py", line 246, in build
    requested_url, discovery_http, cache_discovery, cache, developerKey
  File "~/.local/lib/python3.6/site-packages/googleapiclient/discovery.py", line 308, in _retrieve_discovery_doc
    raise HttpError(resp, content, uri=actual_url)
googleapiclient.errors.HttpError: <HttpError 429 when requesting https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest returned "Too Many Requests">```

**Comment by slgobinath** [comment]
@infomanPMF your error looks different. Google applies some quota limits to the API.
Did you run `gcalendar` several times in a short period? Can you try gcalendar after some time and see if it works?

**Comment by infomanPMF** [comment]
> @infomanPMF your error looks different. Google applies some quota limits to the API.
> Did you run `gcalendar` several times in a short period? Can you try gcalendar after some time and see if it works?

@slgobinath You were right, I had some issues with he calendar and added/removed it too many times which is probably the reason it got blocked. 
It works fine now, thanks

**Comment by slgobinath** [comment]
Fixed with c4851117b1c828a4fad801f4ed8c93351a72a028 and e767395b3091d694b32bfb9ac164a0afb518e1b6

Fix released in [v0.3](https://github.com/slgobinath/gcalendar/releases/tag/v0.3)
